src/components/WalletModal/index.tsx

```diff
@@ -135,105 +130,85 @@ export default function WalletModal({
   confirmedTransactions: string[] // hashes of confirmed
   ENSName?: string
 }) {
-  // important that these are destructed from the account-specific web3-react context
-  const { active, account, connector, activate, error } = useWeb3React()
-
+  const dispatch = useAppDispatch()
+  const { connector, error, hooks } = useWeb3React()
+  const isActiveMap = {
+    [Wallet.INJECTED]: hooks.useSelectedIsActive(injected),
+    [Wallet.COINBASE_WALLET]: hooks.useSelectedIsActive(coinbaseWallet),
+    [Wallet.WALLET_CONNECT]: hooks.useSelectedIsActive(walletConnect),
+  }
   const [walletView, setWalletView] = useState(WALLET_VIEWS.ACCOUNT)
   const previousWalletView = usePrevious(walletView)
 
-  const [pendingWallet, setPendingWallet] = useState<AbstractConnector | undefined>()
-
-  const [pendingError, setPendingError] = useState<boolean>()
+  const [pendingWallet, setPendingWallet] = useState<Connector | undefined>()
 
   const walletModalOpen = useModalOpen(ApplicationModal.WALLET)
   const toggleWalletModal = useWalletModalToggle()
 
-  const previousAccount = usePrevious(account)
-
-  // close on connection, when logged out before
-  useEffect(() => {
-    if (account && !previousAccount && walletModalOpen) {
-      toggleWalletModal()
-    }
-  }, [account, previousAccount, toggleWalletModal, walletModalOpen])
-
-  // always reset to account view
+  const connectorPrevious = usePrevious(connector)
   useEffect(() => {
-    if (walletModalOpen) {
-      setPendingError(false)
+    if (walletModalOpen && connector && connector !== connectorPrevious && !error) {
       setWalletView(WALLET_VIEWS.ACCOUNT)
     }
-  }, [walletModalOpen])
+  }, [setWalletView, error, connector, walletModalOpen, connectorPrevious])
 
-  // close modal when a connection is successful
-  const activePrevious = usePrevious(active)
-  const connectorPrevious = usePrevious(connector)
   useEffect(() => {
-    if (walletModalOpen && ((active && !activePrevious) || (connector && connector !== connectorPrevious && !error))) {
-      setWalletView(WALLET_VIEWS.ACCOUNT)
+    if (connector === network) {
+      setWalletView(WALLET_VIEWS.OPTIONS)
     }
-  }, [setWalletView, active, error, connector, walletModalOpen, activePrevious, connectorPrevious])
+  }, [connector])
 
-  const tryActivation = async (connector: AbstractConnector | undefined) => {
+  const tryActivation = async (connector: Connector) => {
     let name = ''
-    Object.keys(SUPPORTED_WALLETS).map((key) => {
+    Object.keys(SUPPORTED_WALLETS).forEach((key) => {
```

Comment:
forEach is innapropriate here - can you use a for-loop so that it short-circuits.